workitem/workitem_repository.go

```diff
@@ -1131,3 +1161,191 @@ func (r *GormWorkItemRepository) LoadByIteration(ctx context.Context, iterationI
+			fieldDef := oldWIType.Fields[fieldName]
+			oldKind := fieldDef.Type.GetKind()
+			oldValue := fieldDiff[fieldName]
+
+			if oldKind == KindEnum {
+				enumType, ok := fieldDef.Type.(EnumType)
+				if !ok {
+					return errs.Errorf("failed to convert field \"%s\" to enum type: %+v", fieldName, fieldDef)
+				}
+				oldKind = enumType.BaseType.GetKind()
+			}
+			// handle all single value fields (including Enums)
+			if oldKind != KindList {
+				var val string
+				if oldKind.IsRelational() {
+					val, err = getValueOfRelationalKind(r.db, oldValue, oldKind)
+					if err != nil {
+						return errs.Wrapf(err, "failed to get relational value for field %s", fieldName)
+					}
+				} else {
+					val = fmt.Sprint(oldValue)
+				}
+				textToPrepend += fmt.Sprintf("\n\n %s : %s", oldWIType.Fields[fieldName].Label, val)
+				continue
+			}
+
+			// Deal with multi value field (KindList)
+			listType, ok := fieldDef.Type.(ListType)
+			if !ok {
+				return errs.Errorf("failed to convert field \"%s\" to list type: %+v", fieldName, fieldDef)
+			}
+			oldKind = listType.ComponentType.GetKind()
+			valList, ok := fieldDiff[fieldName].([]interface{})
+			if !ok {
+				return errs.Errorf("failed to convert list value of field \"%s\" to []interface{}: %+v", fieldName, fieldDiff[fieldName])
+			}
+
+			var tempList []string
+			for _, v := range valList {
+				val := fmt.Sprint(v)
+				if oldKind.IsRelational() {
+					val, err = getValueOfRelationalKind(r.db, v, oldKind)
+					if err != nil {
+						return errs.Wrapf(err, "failed to get relational value for field %s", fieldName)
+					}
+				}
+				tempList = append(tempList, val)
+			}
+			// Convert []string to comma seperated strings.
+			textToPrepend += fmt.Sprintf("\n\n %s : %s", oldWIType.Fields[fieldName].Label, strings.Join(tempList, ", "))
+		}
+		textToPrepend += "\n\n================================================\n\n"
+		wiStorage.Fields[SystemDescription] = *(rendering.NewMarkupContentFromValue(textToPrepend + originalDescription.Content))
+	}
+	// Set default values for all field in newWIType
+	for fieldName, fieldDef := range newWIType.Fields {
+		fieldValue := wiStorage.Fields[fieldName]
+		// Do not assign default value to metastate
+		if fieldName == SystemMetaState {
+			continue
+		}
+		// Assign default only if fieldValue is nil
+		wiStorage.Fields[fieldName], err = fieldDef.ConvertToModel(fieldName, fieldValue)
+		if err != nil {
+			return errs.Wrapf(err, "failed to convert field %s", fieldName)
+		}
+	}
+	wiStorage.Type = newWIType.ID
+	return nil
+}
+
+// getValueOfRelationKind resolved the relational value stored in val to it's verbose value. Eg: UUID of kind User to username.
+func getValueOfRelationalKind(db *gorm.DB, val interface{}, kind Kind) (string, error) {
+	var result string
+	switch kind {
+	case KindList, KindEnum:
+		return result, errors.NewInternalErrorFromString("cannot resolve relational value for KindList or KindEnum")
```

Comment:
The only reason for why you're returning an error here is because the argument to this function is of type `Kind` and not of type `FieldType`. If it was a `FieldType` you would be able to resolve those list and enum fields that store a relational value.

Reply:
>The only reason for why you're returning an error here is because the argument to this function is of type Kind and not of type FieldType

Lists and Enum have to be dealt with differently. The actual type is stored in `BaseType` and `ComponentType`. That's why we return an error here. The function is supposed to be called with only simple kinds.